builder/builder.go

```diff
@@ -362,11 +363,17 @@ func parseWithdrawalEventAttributes(withdrawalEvent *abcitypes.Event) (*crossdom
 	for _, attr := range withdrawalEvent.Attributes {
 		switch attr.Key {
 		case rolluptypes.AttributeKeySender:
-			senderCosmosAddress, err := sdk.AccAddressFromBech32(attr.Value)
-			if err != nil {
-				return nil, fmt.Errorf("convert sender to cosmos address: %v", err)
+			// check whether the sender address should be encoded as an Ethereum address or a Cosmos address
+			var sender common.Address
+			if common.IsHexAddress(attr.Value) {
+				sender = common.HexToAddress(attr.Value)
```

Comment:
For regular ETH withdrawals, we just use the Cosmos (bech32) address that triggered the withdrawal on L2 as the sender field in the withdrawal tx for proving and finalizing on L1. This is due to us not having the public key associated with that cosmos ETH address, however it shouldn't pose any issues since sender field is only used to calculate the withdrawal hash and is then emitted in an event on L1.

To successfully prove/finalize ERC-20 withdrawals on L1, we need some extra assertions that the sender field for the withdrawal is the [L2CrossDomainMessenger](https://github.com/polymerdao/monomer/blob/nkb.erc-20-withdrawals/x/rollup/keeper/msg_server.go#L157-L158) address. Since this address is static and known at genesis, we don't need to worry about any public key conversions and can just pass through the expected L2CrossDomainMessenger address. The extra check added in the builder is to verify that both types of addresses (cosmos bech32, ETH hex) can be parsed as expected.